utilities/user_utils.py

```
@@ -59,16 +58,15 @@ def create_htpasswd_file(username: str, password: str) -> tuple[Path, str]:
     Returns:
         Tuple of (temp file path, base64 encoded content)
     """
-    with tempfile.NamedTemporaryFile(mode="w+", delete=False) as temp_file:
-        temp_path = Path(temp_file.name).resolve()  # Get absolute path
-        run_command(command=shlex.split(f"htpasswd -c -b {temp_path.absolute()!s} {username} {password}"), check=True)
+    sha1_digest = hashlib.sha1(password.encode()).digest()
```

Ruff (0.15.10)

[error] Probable use of insecure hash functions in `hashlib`: `sha1`

(S324)